starknet_py/net/client_errors.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional, Union
+
+
+class ClientError(Exception):
+    def __init__(self, message: str, code: Optional[str] = None):
+        self.code = code
+        self.message = message
+        super().__init__(self.message)
+
+    def __str__(self):
+        return f"Client failed{f' with code {self.code}' if self.code is not None else ''}: {self.message}"
+
+
+class ContractNotFoundError(ClientError):
+    def __int__(
+        self,
+        block_hash: Optional[Union[int, str]] = None,
+        block_number: Optional[int] = None,
+    ):
+        if block_hash is not None and block_number is not None:
+            raise ValueError(
+                "Block hash and number cannot be provided at the same time"
+            )
+        if block_hash is None and block_number is None:
+            raise ValueError("One of block hash or number must be provided")
```

Comment:
Do we need to check that? In the first case we lose information about the block hash/number by throwing a different error.

Reply:
I haven't consider that, I think I should move this logic out of the constructor then

Reply:
I think we just can take both. In this case `block_hash` will have a priority since `identifier = block_hash or block_number`. This is not a problem since this is something that **we** create, library users won't ever create this error..

Reply:
I guess we can remove this type of checks. I thought it may be good to be explicit that two of these parameters at the same time are not supported.

Reply:
Second ValueError has now been removed. I've retained the first one though, because I think we should actually validate if any of the optional parameters was provided. What do you think?

Reply:
Can we create some helpers for behaviours like this? In here it could be `requireOneOf(NAME_TO_VALUE)`.